### PR TITLE
Upgrade PointCloud to PointCloud2

### PR DESCRIPTION
**Pull request by lukicdarkoo**
**Description**
Here we try to utilize new `getPointCloud(dtype='buffer')` (https://github.com/cyberbotics/webots/pull/2321) to increase performance of the LiDAR.

**Related Issues**
- https://github.com/cyberbotics/webots/issues/2283
- https://answers.ros.org/question/362928/publishreceive-large-pointcloud2-is-very-slow/

**Affected Packages**
  - webots_ros2_core

**Tasks**
  - [x] Add to changelog

**Tests**
```bash
ros2 launch webots_ros2_core robot_launch.py \
    executable:=webots_robotic_arm_node \
    world:=$(ros2 pkg prefix webots_ros2_universal_robot --share)/worlds/universal_robot_lidar.wbt
```

### CODE REVIEW

**Comment by lukicdarkoo** [comment]
With the current implementation, we get a better result (maybe double of speed). However, this is still slow and we have to make it faster.

I benchmarked the implementation and the bottleneck seems to be in ROS2:
```Python
from time import time
import array
import rclpy
from rclpy.node import Node
from std_msgs.msg import Int8MultiArray


MESSAGE_N_BYTES = int(1e6)
PUBLISH_PERIOD = 0.01


class TestPublisher(Node):
    def __init__(self):
        super().__init__('test_publisher')
        self.publisher = self.create_publisher(Int8MultiArray, 'test', 10)
        self.timer = self.create_timer(PUBLISH_PERIOD, self.timer_callback)
        self.data = [0] * MESSAGE_N_BYTES
        self.msg = Int8MultiArray()
        self.msg.data = self.data
        self.data_array = array.array('b', self.data)

    def timer_callback(self):
        tic = time()
        msg = Int8MultiArray()
        msg_ms = (time() - tic) * 1000

        tic = time()
        # msg.data = self.data_array
        msg.data = self.data
        data_ms = (time() - tic) * 1000

        tic = time()
        self.publisher.publish(self.msg)
        pub_ms = (time() - tic) * 1000

        print('Create message', int(msg_ms))
        print('Fill with data', int(data_ms))
        print('Publish message', int(pub_ms))


def main(args=None):
    rclpy.init(args=args)
    test_publisher = TestPublisher()
    rclpy.spin(test_publisher)
    test_publisher.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()
```

Surprisingly, the bottleneck is in the following line:
```
msg.data = self.data
```

as the timings are the following:
```
Fill with data 122
Publish message 10
Create message 0
```

Looking under-the-hood, each message type with the `data` attribute has a setter which converts the input value to Python array:
```Python
    @data.setter
    def data(self, value):
        if isinstance(value, array.array):
            assert value.typecode == 'b', \
                "The 'data' array.array() must have the type code of 'b'"
            self._data = value
            return
        if __debug__:
            from collections.abc import Sequence
            from collections.abc import Set
            from collections import UserList
            from collections import UserString
            assert \
                ((isinstance(value, Sequence) or
                  isinstance(value, Set) or
                  isinstance(value, UserList)) and
                 not isinstance(value, str) and
                 not isinstance(value, UserString) and
                 all(isinstance(v, int) for v in value) and
                 all(val >= -128 and val < 128 for val in value)), \
                "The 'data' field must be a set or sequence and each value of type 'int' and each integer in [-128, 127]"
        self._data = array.array('b', value)
```

This conversion (`self._data = array.array('b', value)`) actually takes the most of the time.

In conclusion, we can return the correct array type directly from Webots and in that way avoid the ROS2 bottleneck. Working on that...

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -6,7 +6,6 @@ on:
 
 jobs:
   industrial_ci:
-    if: github.event.pull_request.draft == false
```
Comment (omichel) [inline .github/workflows/test.yml:9]: Is that on purpose? I would assume yes and it's fine, but I just want to be sure you are aware of it.

File: webots_ros2_core/CHANGELOG.rst [hunk 2]
```diff
@@ -2,6 +2,10 @@
 Changelog for package webots_ros2_core
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+1.0.3 (2020-xx-xx)
+------------------
+* Improved point cloud publishing by a few times.
```
Comment (omichel) [inline webots_ros2_core/CHANGELOG.rst:7]:
```suggestion
* Improved the performance of point cloud publishing by a few times.
```

**Review by omichel — COMMENTED** [review]
Very good.

File: .github/workflows/test.yml (hunk 1 above)
Comment (lukicdarkoo) [reply .github/workflows/test.yml:9]: Yes. That line doesn't make sense for me, as I want to see if my CI is failing before I request a review. It is already deleted in the master branch (I will master to develop later).

**Comment by lukicdarkoo** [comment]
I fixed the warning from DeepCode

File: webots_ros2_core/webots_ros2_core/devices/lidar_device.py [hunk 3]
```diff
@@ -92,12 +91,28 @@ def step(self):
             self._wb_device.disable()
 
     def __publish_point_cloud_data(self, stamp):
-        points = self._wb_device.getPointCloud()
-        if points:
-            msg = PointCloud()
+        data = self._wb_device.getPointCloud(dtype='buffer')
```
Comment (omichel) [inline webots_ros2_core/webots_ros2_core/devices/lidar_device.py:94]:
```suggestion
        data = self._wb_device.getPointCloud(data_type='buffer')
```

**Review by omichel — CHANGES_REQUESTED** [review]

File: webots_ros2_core/webots_ros2_core/devices/lidar_device.py (hunk 3 above)
Comment (lukicdarkoo) [reply webots_ros2_core/webots_ros2_core/devices/lidar_device.py]: Thanks!

**Review by omichel — APPROVED** [review]
Thank you.
